autogen/agentchat/contrib/rag/rag_agent.py

```diff
@@ -0,0 +1,810 @@
+        else:
+            raise ValueError(f"Invalid retriever: {self._retriever}.")
+
+        # initialize the reranker
+        self._reranker = self.rag_config.get("reranker", "tfidf")
+        if isinstance(self._reranker, str):
+            self.reranker = RerankerFactory.create_reranker(self._reranker)
+        elif isinstance(self._reranker, Reranker):
+            self.reranker = self._reranker
+        else:
+            raise ValueError(f"Invalid reranker: {self._reranker}.")
+
+        # Chunk data and Create collection
+        self.overwrite = self.rag_config.get("overwrite", False)
+        self.get_or_create = self.rag_config.get("get_or_create", True)
+        self.upsert = self.rag_config.get("upsert", True)
+
+        if not self.splitter.docs_path:
+            try:
+                self.retriever.vector_db.get_collection(self.collection_name)
+                logger.warning(
+                    "`docs_path` is not provided for splitter. "
+                    f"Use the existing collection `{self.collection_name}`.",
+                    color="yellow",
+                )
+                self.overwrite = False
+                self.get_or_create = True
+                _docs = None
+            except ValueError:
+                raise ValueError(
+                    "`docs_path` is not provided for splitter. "
+                    f"The collection `{self.collection_name}` doesn't exist either. "
+                    "Please provide `docs_path` or create the collection first."
+                )
+        elif self.get_or_create and not self.overwrite:
+            try:
+                self.retriever.vector_db.get_collection(self.collection_name)
+                logger.info(f"Use the existing collection `{self.collection_name}`.", color="green")
+                _docs = None
+            except ValueError:
+                # Chunk raw files
+                _chunks = self.splitter.split()
+                _docs = self.encoder.encode_chunks(_chunks)
+        else:
+            # Chunk raw files
+            _chunks = self.splitter.split()
+            _docs = self.encoder.encode_chunks(_chunks)
+        self.retriever.vector_db.create_collection(self.collection_name, self.overwrite, self.get_or_create)
+        self.retriever.vector_db.insert_docs(_docs, self.collection_name, upsert=self.upsert)
+
+        # initialize the inner agents
+        inner_llm_config = copy.deepcopy(llm_config)
+        self._assistant = AssistantAgent(
+            self.name + "_inner_assistant",
+            system_message=system_message,
+            llm_config=inner_llm_config,
+            is_termination_msg=lambda m: False,
+        )
+
+        self._user_proxy = UserProxyAgent(
+            self.name + "_inner_user_proxy",
+            human_input_mode="NEVER",
+            code_execution_config=False,
+            default_auto_reply="",
+            is_termination_msg=lambda m: False,
+        )
+
+        self.ipython = get_ipython()
```

Comment:
We currently have Jupyter code executor and adding the containerized version of it. That would be a more secure version of running code directly in the embedded ipython kernel. https://github.com/microsoft/autogen/pull/1794